ui/lib/core/addon/components/search-select-with-modal.js

```diff
@@ -0,0 +1,217 @@
+
+/**
+ * @module SearchSelectWithModal
+ * The `SearchSelectWithModal` is an implementation of the [ember-power-select](https://github.com/cibernox/ember-power-select) used for form elements where options come dynamically from the API. It can only accept a single model.
+ * It renders a passed in form component so records can be created inline, via a modal that pops up after clicking "Add new <record>" from the dropdown menu
+ * @example
+ * <SearchSelectWithModal
+ *         @id="assignments"
+ *         @label="assignment"
+ *         @labelClass="is-label"
+ *         @subText="Search for an existing assignment, or type a new name to create it."
+ *         @model="oidc/assignment"
+ *         @inputValue={{map-by "id" @model.assignments}}
+ *         @onChange={{this.handleSearchSelect}}
+ *         {{! since this is the "limited" radio select option we do not want to include 'allow_all' }}
+ *         @excludeOptions={{array "allow_all"}}
+ *         @fallbackComponent="string-list"
+ *         @modalFormComponent="oidc/assignment-form"
+ *         @modalSubtext="Use assignment to specify which Vault entities and groups are allowed to authenticate."
+ *       />
+ *
+ * @param {string} id - The name of the form field
+ * @param {Array} model - model type to fetch from API (can only be a single model)
+ * @param {function} onChange - The onchange action for this form field. ** SEE UTIL ** search-select-has-many.js if selecting models from a hasMany relationship
+ * @param {string | Array} inputValue -  A comma-separated string or an array of strings -- array of ids for models.
+ * @param {string} label - Label for this form field
+ * @param {string} fallbackComponent - name of component to be rendered if the API returns a 403s
+ * @param {boolean} [passObject=false] - When true, the onChange callback returns an array of objects with id (string) and isNew (boolean)
+ * @param {string} [helpText] - Text to be displayed in the info tooltip for this form field
+ * @param {number} [selectLimit] - A number that sets the limit to how many select options they can choose
+ * @param {string} [subText] - Text to be displayed below the label
+ * @param {string} [subLabel] - a smaller label below the main Label
+ * @param {string} [placeholder] - placeholder text to override the default text of "Search"
+ * @param {array} [excludeOptions] - array of strings containing model ids to filter from the dropdown (ex: ['allow_all'])
+ * @param {function} search - *Advanced usage* - Customizes how the power-select component searches for matches -
+ * see the power-select docs for more information.
+ *
+ */
+export default class SearchSelectWithModal extends Component {
+  @service store;
+
+  @tracked selectedOptions = null; // list of selected options
+  @tracked allOptions = null; // all possible options
+  @tracked showModal = false;
+  @tracked newModelRecord = null;
+  @tracked shouldUseFallback;
```

Comment:
This should be false by default yes?

Reply:
yes - thank you! I deleted the default by accident